File: src/main/java/fi/quanfoxes/lexer/ClassicOperator.java
package fi.quanfoxes.lexer;

public class ClassicOperator extends Operator {
    private boolean shared;

    public ClassicOperator(String identifier, int priority) {
        this(identifier, priority, true);
    }

    public ClassicOperator(String identifier, int priority, boolean shared) {
        super(identifier, OperatorType.CLASSIC, priority);
        this.shared = shared;
    }

    public boolean isSharedContext() {
        return shared;
    }
}

File: Vivid/Lexer/ClassicOperator.cs
using System;

public class ClassicOperator : Operator
{
	public bool IsShared { get; private set; }

	public ClassicOperator(string identifier, int priority, bool shared = true) : base(identifier, OperatorType.CLASSIC, priority)
	{
		IsShared = shared;
	}

	public override bool Equals(object? other)
	{
		return other is ClassicOperator operation &&
			   base.Equals(other) &&
			   IsShared == operation.IsShared;
	}

	public override int GetHashCode()
	{
		return HashCode.Combine(base.GetHashCode(), IsShared);
	}
}